Server error in Connect command:  [9]

Aerospike Community Edition build 4.3.0.2 (Docker)

```
$ docker exec -it db aql
Seed:         127.0.0.1
User:         None
Config File:  /etc/aerospike/astools.conf /root/.aerospike/astools.conf
Aerospike Query Client
Version 3.15.3.14
C Client Version 4.3.12
Copyright 2012-2017 Aerospike. All rights reserved.
aql> show namespaces;
+------------+
| namespaces |
+------------+
| "test"     |
+------------+
[127.0.0.1:3000] 1 row in set (0.002 secs)

OK

aql>
```

```bash
$ ncat localhost 3000 -v
Ncat: Version 7.70 ( https://nmap.org/ncat )
Ncat: Connected to ::1:3000.
^C
```

```javascript
let client = Aerospike.client({
  hosts: [
    { addr: "127.0.0.1", port: 3000 }
  ],
  log: {
    level: Aerospike.log.TRACE
  }
})

client.connect((error) => {
  if (error) {
    console.error('Connection to Aerospike cluster failed!', error);
    process.exit(-1);
  }
});
```

```
npm run start

> api@1.0.0 start /path/to/my/code/api
> node --experimental-modules index.mjs

(node:24368) ExperimentalWarning: The ESM module loader is experimental.
Aug 06 2018 16:56:16 UTC: TRACE(24368) [conversions.cc:181] [get_optional_uint32_property] - port => (uint32) 3000
Aug 06 2018 16:56:16 UTC: TRACE(24368) [conversions.cc:184] [get_optional_uint32_property] - authMode => undefined
Aug 06 2018 16:56:16 UTC: TRACE(24368) [config.cc:82] [config_from_jsobject] - adding host, addr="127.0.0.1", port=3000
Aug 06 2018 16:56:16 UTC: DEBUG(24368) [config.cc:161] [config_from_jsobject] - Parsing global policies: success
Aug 06 2018 16:56:16 UTC: TRACE(24368) [conversions.cc:85] [get_optional_string_property] - systemPath => "/path/to/my/code/api/node_modules/aerospike/modules/lua-core/src"
Aug 06 2018 16:56:16 UTC: DEBUG(24368) [config.cc:185] [config_from_jsobject] - Using default Lua user path: /usr/local/aerospike/client/usr/udf/lua
Aug 06 2018 16:56:16 UTC: TRACE(24368) [conversions.cc:184] [get_optional_uint32_property] - connTimeoutMs => undefined
Aug 06 2018 16:56:16 UTC: TRACE(24368) [conversions.cc:184] [get_optional_uint32_property] - loginTimeoutMs => undefined
Aug 06 2018 16:56:16 UTC: TRACE(24368) [conversions.cc:184] [get_optional_uint32_property] - tendInterval => undefined
Aug 06 2018 16:56:16 UTC: TRACE(24368) [conversions.cc:184] [get_optional_uint32_property] - maxConnsPerNode => undefined
Aug 06 2018 16:56:16 UTC: TRACE(24368) [conversions.cc:184] [get_optional_uint32_property] - maxConnsPerNodeSync => undefined
Aug 06 2018 16:56:16 UTC: DEBUG(24368) [client.cc:82] [New] - Aerospike client initialized successfully
Aug 06 2018 16:56:16 UTC: DEBUG(24368) [events.cc:52] [push] - Cluster event 0 triggered by node "BB9020015AC4202" (172.21.0.2:3000)
Aug 06 2018 16:56:17 UTC: DEBUG(24368) [command.cc:86] [ErrorCallback] - Server error in Connect command:  [9]
Aug 06 2018 16:56:17 UTC: DEBUG(24368) [command.cc:66] [Callback] - Executing JS callback for Connect command
Aug 06 2018 16:56:17 UTC: DEBUG(24368) [client.cc:116] [Close] - Closing the connection to aerospike cluster
Connection to Aerospike cluster failed! { AerospikeError
    at Function.fromASError (/path/to/my/code/api/node_modules/aerospike/lib/error.js:113:19)
    at as_client.connect (/path/to/my/code/api/node_modules/aerospike/lib/client.js:508:28)
    at Client.connect (/path/to/my/code/api/node_modules/aerospike/lib/client.js:507:20)
    at file:///path/to/my/code/api/db/index.mjs:32:8
    at ModuleJob.run (internal/modules/esm/module_job.js:96:12)
  name: 'AerospikeError',
  code: 9,
  command: null,
  func: '',
  file: '',
  line: 0,
  inDoubt: false }
npm ERR! code ELIFECYCLE
npm ERR! errno 255
npm ERR! api@1.0.0 start: `node --experimental-modules index.mjs`
npm ERR! Exit status 255
npm ERR!
npm ERR! Failed at the api@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     ~/.npm/_logs/2018-08-06T16_56_17_447Z-debug.log
```

`package.json -> "aerospike": "^3.5.0"`

Connection is fine, but it has some issue with `Server error in Connect command:  [9]`.

P.S. Just checked with v3.4.0 - works fine. Looks like regression with nodejs-client v.3.5.0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,9 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+* **Bug Fixes**
+  * Support using alternate-access-address. [#267](https://github.com/aerospike/aerospike-client-nodejs/issue/267)
+
 ## [3.7.0] - 2018-10-02
 
 * **New Features**

--- a/lib/config.js
+++ b/lib/config.js
@@ -368,6 +368,18 @@ class Config {
     if (typeof config.sharedMemory === 'object') {
       this.sharedMemory = config.sharedMemory
     }
+
+    /**
+     * @name Config#useAlternateAccessAddress
+     * @summary Whether the client should use the server's
+     * <code>alternate-access-address</code> instead of the
+     * <code>access-address</code>.
+     *
+     * @type {boolean}
+     * @default false
+     * @since v3.7.1
+     */
+    this.useAlternateAccessAddress = Boolean(config.useAlternateAccessAddress)
   }
 
   /**

--- a/src/main/config.cc
+++ b/src/main/config.cc
@@ -230,6 +230,9 @@ int config_from_jsobject(as_config* config, Local<Object> configObj, const LogIn
 	if ((rc = get_optional_uint32_property(&config->max_conns_per_node, NULL, configObj, "maxConnsPerNodeSync", log)) != AS_NODE_PARAM_OK) {
 		goto Cleanup;
 	}
+	if ((rc = get_optional_bool_property(&config->use_services_alternate, NULL, configObj, "useAlternateAccessAddress", log)) != AS_NODE_PARAM_OK) {
+		goto Cleanup;
+	}
 
 Cleanup:
 	if (cluster_name) free(cluster_name);

